### PR TITLE
osu! requests and other tweaks

### PR DESCRIPTION
**Pull request by n2468txd**
New module: osu! requests

Other fixes:
- Big Black in #kano
- Twitchbot IRC simplification in config (remove host and port and hardcode them)
- Clear up 'Logged in to x server' (make x to tell which client)

### CODE REVIEW

File: lib/osu/osu_requests.ex [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+      # Check that the request isn't invalid (i.e. nothing in data)
+      if beatmaps_data != [] do
+        # Get first element in the JSON data (i.e. the hd()) as the hardest diff is usually the first one
+        map_data = hd(beatmaps_data)
+        map_artist = Map.get(map_data, "artist")
+        map_title = Map.get(map_data, "title")
+        map_diff = Map.get(map_data, "version")
+        map_BPM = Map.get(map_data, "bpm")
+        map_star = Float.round(elem((Float.parse(Map.get(map_data, "difficultyrating"))), 0), 2) # round to 2 decimal places
```
Comment (michaelowens) [inline lib/osu/osu_requests.ex]: `map_data |> Map.get("difficultyrating") |> Float.parse |> elem(0) |> Float.round(2)`

That's probably nicer and easier to follow

File: lib/osu/osu_requests.ex [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  # Catch-all for messages you don't care about
+  def handle_info(_msg, state) do
+    {:noreply, state}
+  end
+
+  defp debug(msg) do
+    IO.puts IO.ANSI.yellow() <> msg <> IO.ANSI.reset()
+  end
+
+  def handle_osu_request({channel, user, type, id}, client) do
```
Comment (michaelowens) [inline lib/osu/osu_requests.ex]: This function seems kind of big, maybe split it up in a couple of smaller functions that do a specific thing.

File: lib/osu/osu_requests.ex [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    GenServer.start_link(__MODULE__, [client])
+  end
+
+  def init([client]) do
+    ExIrc.Client.add_handler client, self
+    {:ok, client}
+  end
+
+  def handle_info({:joined, channel}, client) do
+    debug "Joined #{channel}"
```
Comment (michaelowens) [inline lib/osu/osu_requests.ex]: Maybe we should add something to `debug/1`, or at least to these messages so we can see which connection logged this. Kind of like you also added twitch/bancho to the connection message so we can distinguish them.

File: lib/osu/osu_requests.ex (hunk 2 above)
Comment (n2468txd) [inline lib/osu/osu_requests.ex]: Uhh not sure which functions you mean but I've cleaned up some code.

Comment (michaelowens) [inline lib/osu/osu_requests.ex]: The line I commented on :p handle_osu_request

Comment (n2468txd) [inline lib/osu/osu_requests.ex]: I honestly don't know how I could split it up, it's all just dedicated to handling osu! requests.

... lmao

Comment (michaelowens) [inline lib/osu/osu_requests.ex]: It's not too bad actually, I was looking on my phone when I sent those messages and it looked worse on phone haha

**Pull request closed without merging**